### ISSUE TITLE
Pods paused on `AppArmor` while trying to deploy

### ISSUE THREAD

**Issue by weirdwiz** [kubearmor/KubeArmor#283]
On Kubernetes v1.22.1, pods deployed display the following error.

```
AppArmor
Cannot enforce AppArmor: profile "kubearmor-[namespace]-[podname]" is not loaded
````
But enforcement still works, and the policy is listed when checking `/etc/apparmor.d`.

**Comment by nyrahul** [comment]
More weirdly, in some situations, the status differs between similar pods. In the above case, we have Ubuntu-2 showing "Running" and Ubuntu-1 showing "AppArmor".

![image](https://user-images.githubusercontent.com/9133227/130597142-be390b45-3c12-4a02-afc9-fd7b1ab705cc.png)

<details>
  <summary>kubectl describe pod -n multiubuntu ubuntu-1-deployment-5d6b975744-8fptw</summary>
```
Name:         ubuntu-1-deployment-5d6b975744-8fptw
Namespace:    multiubuntu
Priority:     0
Node:         vbox1804/192.168.122.57
Start Time:   Tue, 24 Aug 2021 09:31:46 +0000
Labels:       container=ubuntu-1
              group=group-1
              pod-template-hash=5d6b975744
Annotations:  container.apparmor.security.beta.kubernetes.io/ubuntu-1-container: localhost/kubearmor-multiubuntu-ubuntu-1-container
              kubearmor-policy: enabled
              kubearmor-visibility: process,file,network
Status:       Running
Reason:       AppArmor
Message:      Cannot enforce AppArmor: profile "kubearmor-multiubuntu-ubuntu-1-container" is not loaded
IP:           10.0.0.158
IPs:
  IP:           10.0.0.158
Controlled By:  ReplicaSet/ubuntu-1-deployment-5d6b975744
Containers:
  ubuntu-1-container:
    Container ID:   docker://76564c929c2ff71943d461e924aaf302c76e3cf86964570235c158a48a7a0051
    Image:          kubearmor/ubuntu-w-utils:0.1
    Image ID:       docker-pullable://kubearmor/ubuntu-w-utils@sha256:73a3513de2f71c5fb977ca7b92c115fbe5cd938d0afdffed7a88b9a8ff0a4a85
    Port:           <none>
    Host Port:      <none>
    State:          Running
      Started:      Tue, 24 Aug 2021 09:32:41 +0000
    Ready:          True
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-kdc2w (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
Volumes:
  kube-api-access-kdc2w:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type    Reason     Age   From               Message
  ----    ------     ----  ----               -------
  Normal  Scheduled  27m   default-scheduler  Successfully assigned multiubuntu/ubuntu-1-deployment-5d6b975744-8fptw to vbox1804
  Normal  Pulling    27m   kubelet            Pulling image "kubearmor/ubuntu-w-utils:0.1"
  Normal  Pulled     26m   kubelet            Successfully pulled image "kubearmor/ubuntu-w-utils:0.1" in 40.18050037s
  Normal  Created    26m   kubelet            Created container ubuntu-1-container
  Normal  Started    26m   kubelet            Started container ubuntu-1-container
```
</details>

<details>
  <summary>kubectl describe pod -n multiubuntu ubuntu-2-deployment-bc58cf44-w859k</summary>
```
Name:         ubuntu-2-deployment-bc58cf44-w859k
Namespace:    multiubuntu
Priority:     0
Node:         vbox1804/192.168.122.57
Start Time:   Tue, 24 Aug 2021 09:31:46 +0000
Labels:       container=ubuntu-2
              group=group-1
              pod-template-hash=bc58cf44
Annotations:  container.apparmor.security.beta.kubernetes.io/ubuntu-2-container: localhost/kubearmor-multiubuntu-ubuntu-2-container
              kubearmor-policy: enabled
Status:       Running
IP:           10.0.0.238
IPs:
  IP:           10.0.0.238
Controlled By:  ReplicaSet/ubuntu-2-deployment-bc58cf44
Containers:
  ubuntu-2-container:
    Container ID:   docker://53011ceec7ead9db19e54821058a2224b585469a10309ae1b879780e5c44563f
    Image:          kubearmor/ubuntu-w-utils:0.1
    Image ID:       docker-pullable://kubearmor/ubuntu-w-utils@sha256:73a3513de2f71c5fb977ca7b92c115fbe5cd938d0afdffed7a88b9a8ff0a4a85
    Port:           <none>
    Host Port:      <none>
    State:          Running
      Started:      Tue, 24 Aug 2021 09:32:52 +0000
    Ready:          True
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-fn5vn (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
Volumes:
  kube-api-access-fn5vn:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:                      <none>
```
</details>

**Comment by oneiro-naut** [comment]
Possibly related https://github.com/kubernetes/kubernetes/pull/96364.

**Comment by oneiro-naut** [comment]
@nyrahul @nam-jaehyun This bug seems to be happening for usual deployments having the apparmor annotations(I used the simple hello-apparmor example from k8s documentation and it gives exact same behavior as this issue). I have identified the steps to produce it. The bug occurs whenever a pod is deployed but at the time of it's creation the specified profile is not loaded. When we later load it the status changes and the pod starts running. Now at this point the actual status is `Running` but the `kubectl get pods` command shows `Apparmor` and in the describe command we still see `Cannot enforce AppArmor: profile X is not loaded`. Note that at this point the profile is actually there. So basically the `Reason` and `Message` shown are not being updated for some reason. I tried the same on k8s version 1.21 but it didn't happen.
Example from k8s version 1.22:
```
vagrant@ubuntu-focal:~$ kubectl describe po
Name:         hello-apparmor
Namespace:    default
Priority:     0
Node:         ubuntu-focal/10.0.2.15
Start Time:   Thu, 09 Sep 2021 07:28:39 +0000
Labels:       <none>
Annotations:  container.apparmor.security.beta.kubernetes.io/hello: localhost/k8s-apparmor-example-deny-write
Status:       Running
Reason:       AppArmor
Message:      Cannot enforce AppArmor: profile "k8s-apparmor-example-deny-write" is not loaded
``` 
In the kubearmor the bug is triggered during rolling update to add apparmor annotations. During that phase we have containers already created(since we got ADDED event in the k8sHandler) after that they are patched 
https://github.com/kubearmor/KubeArmor/blob/ac6861442e2e7f1c6b04d67bd79b1fe6526573cb/KubeArmor/core/kubeUpdate.go#L494
and then later when we receive event for these patched pods the corresponding profile is registered.
https://github.com/kubearmor/KubeArmor/blob/ac6861442e2e7f1c6b04d67bd79b1fe6526573cb/KubeArmor/core/kubeUpdate.go#L625
Note that there is a time interval here when we have pods but the corresponding profiles are yet to be registered(this even happens in the version 1.21). And I think this triggers this k8s bug in which the `Reason` is not updated and wrong `Status` is shown.
